eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Mgmt/src/Providers/ResourceClientProvider.cs

```diff
@@ -239,58 +242,91 @@ private MethodProvider BuildOperationMethod(InputServiceMethod method, MethodPro
                 {
-                    UsingDeclare("scope", typeof(DiagnosticScope), _clientDiagonosticsField.Invoke(nameof(ClientDiagnostics.CreateScope), [Literal($"{Type.Namespace}.{operation.Name}")]), out var scopeVariable),
+                    UsingDeclare("scope", typeof(DiagnosticScope), _clientDiagonosticsField.Invoke(nameof(ClientDiagnostics.CreateScope), [Literal($"{Name}.{signature.Name}")]), out var scopeVariable),
                     scopeVariable.Invoke(nameof(DiagnosticScope.Start)).Terminate(),
                     new TryCatchFinallyStatement
-                    (BuildOperationMethodTryStatement(convenienceMethod, isAsync, method, isGeneric), Catch(Declare<Exception>("e", out var exceptionVarialble), [scopeVariable.Invoke(nameof(DiagnosticScope.Failed), exceptionVarialble).Terminate(), Throw()]))
+                    (BuildOperationMethodTryStatement(method, convenienceMethod, signature, isAsync, isGeneric),
+                     Catch(Declare<Exception>("e", out var exceptionVarialble),
+                     [scopeVariable.Invoke(nameof(DiagnosticScope.Failed), exceptionVarialble).Terminate(), Throw()]))
                 };
 
             return new MethodProvider(signature, bodyStatements, this);
         }
 
+        protected virtual bool SkipMethodParameter(ParameterProvider parameter)
+        {
+            return ContextualParameters.Contains(parameter.Name);
+        }
+
         protected IReadOnlyList<ParameterProvider> GetOperationMethodParameters(MethodProvider convenienceMethod, bool isLongRunning)
         {
             var result = new List<ParameterProvider>();
             if (isLongRunning)
             {
                 result.Add(KnownAzureParameters.WaitUntil);
             }
+
             foreach (var parameter in convenienceMethod.Signature.Parameters)
             {
-                if (!_contextualParameters.Contains(parameter.Name))
+                if (!SkipMethodParameter(parameter))
                 {
                     result.Add(parameter);
                 }
             }
+
             return result;
         }
 
-        protected CSharpType GetOperationMethodReturnType(bool isAsync, bool isLongRunningOperation, IReadOnlyList<InputOperationResponse> operationResponses, out bool isGeneric)
+        protected bool IsLongRunningOperation(InputServiceMethod method)
+        {
+            return method is InputLongRunningServiceMethod || method is InputLongRunningPagingServiceMethod;
+        }
+
+        protected bool IsReturnTypeGeneric(InputServiceMethod method)
+        {
+            var operationResponses = method.Operation.Responses;
+            var response = operationResponses.FirstOrDefault(r => !r.IsErrorResponse);
+            var responseBodyType = response?.BodyType is null ? null : ManagementClientGenerator.Instance.TypeFactory.CreateCSharpType(response.BodyType);
+            return IsLongRunningOperation(method) && responseBodyType is not null;
+        }
```

Comment:
similar here I think this should be an extension method for InputServiceMethod, or in an internal helper class.
Also I think we should make this method to return the body type as a helper.
I think the purpose of this method is to determine whether the generated method should use `ArmOperation` or `ArmOperation<T>` as its return type, but this is decided by whether the long running operation has the body.

Therefore we should make a helper to return the body type, and on the caller side, we could check if this is long running, and if the body type is present.